### PR TITLE
hpccm: init at 22.5.0

### PR DESCRIPTION
**Pull request by AtilaSaraiva**
###### Description of changes

HPC Container Maker is an application that helps create Nvidia compatible docker (and singularity) containers.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>hpccm</li>
    <li>python310Packages.archspec</li>
    <li>python39Packages.archspec</li>
  </ul>
</details>

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/archspec/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  nativeBuildInputs = [ poetry-core ];
+  propagatedBuildInputs = [ click six ];
+  checkInputs = [ pytestCheckHook jsonschema ];
+
+  pythonImportsCheck = [ "archspec" ];
+
+  meta = with lib; {
+    description = "A library for detecting, labeling, and reasoning about microarchitectures";
+    homepage = "archspec.readthedocs.io/";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/archspec/default.nix:32]: This must be a valid URL.

File: pkgs/development/python-modules/archspec/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  pname = "archspec";
+  version = "0.1.4";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-ScigEpYNArveqi5tlqiA7LwsVs2RkjT+GChxhSy/ndw=";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/archspec/default.nix:21]:
```suggestion
    fetchSubmodules = true;
    sha256 = "sha256-ScigEpYNArveqi5tlqiA7LwsVs2RkjT+GChxhSy/ndw=";
```

File: pkgs/applications/virtualization/hpccm/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    repo = "hpc-container-maker";
+    rev = "v${version}";
+    sha256 = "sha256-zR5+X9BKaUvLPQ05FnfU817esgxVqP8n+wfdWy20BN4=";
+  };
+
+  propagatedBuildInputs = [ six archspec ];
+  checkInputs = [ pytestCheckHook pytest-xdist ];
+
+  # These two tests require git
+  disabledTests = [ "test_commit" "test_tag" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/hpccm/default.nix:25]:
```suggestion
  disabledTests = [
    # tests require git
    "test_commit"
    "test_tag"
  ];
```

File: pkgs/development/python-modules/archspec/default.nix (hunk 1 above)
Comment (AndersonTorres) [inline pkgs/development/python-modules/archspec/default.nix:32]:
```suggestion
    homepage = "https://archspec.readthedocs.io/en/latest/";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/archspec/default.nix (hunk 1 above)
Comment (AtilaSaraiva) [reply pkgs/development/python-modules/archspec/default.nix]: Thank you for the review as always @SuperSandro2000 , I really appreciate it. 

Done!

File: pkgs/development/python-modules/archspec/default.nix (hunk 2 above)
Comment (AtilaSaraiva) [reply pkgs/development/python-modules/archspec/default.nix]: Done!

File: pkgs/development/python-modules/archspec/default.nix (hunk 1 above)
Comment (AtilaSaraiva) [reply pkgs/development/python-modules/archspec/default.nix]: Thanks for the review Anderson! Done.

**Comment by AtilaSaraiva** [comment]
Added support for the python module version.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>hpccm (python39Packages.hpccm)</li>
    <li>python310Packages.archspec</li>
    <li>python310Packages.hpccm</li>
    <li>python39Packages.archspec</li>
  </ul>
</details>

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -4009,6 +4011,8 @@ in {
 
   hpack = callPackage ../development/python-modules/hpack { };
 
+  hpccm = callPackage ../development/python-modules/hpccm { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:4014]: If this is an application, not a python library please remove this entry.

Comment (AtilaSaraiva) [reply pkgs/top-level/python-packages.nix:4014]: Hello Sandro, this is both a python application and package.

**Comment by AndersonTorres** [comment]
@AtilaSaraiva is this package expected to run in ARM machines?

**Comment by AtilaSaraiva** [comment]
> @AtilaSaraiva is this package expected to run in ARM machines?

It would be interesting to see, but I can't see why not. Afaik the program is pure python, and it only creates a Dockerfile (or Singularity.def).

**Comment by AndersonTorres** [comment]
> > @AtilaSaraiva is this package expected to run in ARM machines?
> 
> It would be interesting to see, but I can't see why not. Afaik the program is pure python, and it only creates a Dockerfile (or Singularity.def).

Well, ofborg is complaining. Dunno if related, tho.

**Comment by AtilaSaraiva** [comment]
> > > @AtilaSaraiva is this package expected to run in ARM machines?
> > 
> > It would be interesting to see, but I can't see why not. Afaik the program is pure python, and it only creates a Dockerfile (or Singularity.def).
> 
> Well, ofborg is complaining. Dunno if related, tho.

Oh, I didn't see that. Lemme check.

**Comment by SuperSandro2000** [comment]
It does not support arm:

```
        if hpccm.config.g_cpu_arch == cpu_arch.PPC64LE:
            self.__arch_directory = 'linuxpower'
            self.__arch_pkg = 'openpower'
        elif hpccm.config.g_cpu_arch == cpu_arch.X86_64:
            self.__arch_directory = 'linux86-64'
            self.__arch_pkg = 'x64'
        else: # pragma: no cover
>           raise RuntimeError('Unknown CPU architecture')
E           RuntimeError: Unknown CPU architectu
```

**Comment by AtilaSaraiva** [comment]
huh, interesting, thanks for finding out!

File: pkgs/development/python-modules/hpccm/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    "test_tag"
+  ];
+
+  pythonImportsCheck = [ "hpccm" ];
+
+  meta = with lib; {
+    description = "HPC Container Maker";
+    homepage = "https://github.com/NVIDIA/hpc-container-maker";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ atila ];
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/hpccm/default.nix:36]: That being said, add a suitable `badPlatforms`.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Pull request merged**